source/src/python/PyRosetta/src/pyrosetta/distributed/cluster/core.py

```diff
@@ -118,8 +119,11 @@
     pyrosetta_build: A `str` or `NoneType` object specifying the PyRosetta build as
         output by `pyrosetta._version_string()`. If `None` is provided, then PyRosettaCluster
         automatically detects the PyRosetta build and sets this attribute as the `str`.
-        If a `str` is provided, then validate that the input PyRosetta build is equal
-        to the active PyRosetta build, and log a warning message if not.
+        If a non-empty `str` is provided, then validate that the input PyRosetta build is
+        equal to the active PyRosetta build, and raise an error if not. This ensures that
+        reproduction simulations use an identical PyRosetta build from the original
+        simulation. To bypass PyRosetta build validation with a warning message, an
+        empty string ('') may be provided (but does not ensure reproducibility).
```

Comment:
@klimaj what value exactly is used when checking if PyRosetta version is equal? Is that output of `pyrosetta._version_string()`? I am asking because technically that not enough to guarantee that versions are equal. More precise would be to use both version string and output of `pyrosetta.rosetta.utility.Version.extras()` (probably sorted and concatenated into a string). That way we can distinguish slightly different build flavors that could yield difference down the line, like builds with different extras.

Reply:
@lyskov Oh good catch, thanks! Indeed we are just comparing `pyrosetta._version_string()` here (which is probably enough to distinguish the `pyrosetta_installer` PyPI package options `serialization=True/False` (which puts/omits `cxx11thread.serialization` to/from the `pyrosetta._version_string()`) and the `type='Release'/'MinSizeRel'` (also in the `pyrosetta._version_string()`). Still that's correct- I suppose a user could also do a custom PyRosetta install into a conda environment with some extras that would not be caught. So yes, I agree, perhaps we can add a `pyrosetta._complete_version_string()` or similar in PyRosetta (in a separate PR, since it's a bit out-of-scope of this PR)